### PR TITLE
RYA-200 Removing findbugs:jsr305 Dependency

### PR DESCRIPTION
**Pull request by amihalik**
## Description

Removed findbugs:jsr305 Dependency.  Replaced with findbugs-annotations.

Several annotations are not available in findbugs-annotations.  These annotations where commented out and a new JIRA ticket (RYA-211) was created to deal with them:

- @ThreadSafe
- @Immutable
- @ParametersAreNonnullByDefault

### Tests

Current tests pass.  No New tests.

### Links
[Jira](https://issues.apache.org/jira/browse/RYA-200)

### Checklist
- [ ] Code Review
- [ ] Squash Commits

#### People To Reivew
@isper3at

### CODE REVIEW

File: extras/rya.console/src/main/java/org/apache/rya/shell/SharedShellState.java [hunk 1]
```diff
@@ -23,10 +23,10 @@
 import java.util.Objects;
 import java.util.concurrent.locks.ReentrantLock;
 
-import javax.annotation.Nullable;
-import javax.annotation.ParametersAreNonnullByDefault;
-import javax.annotation.concurrent.Immutable;
-import javax.annotation.concurrent.ThreadSafe;
+import edu.umd.cs.findbugs.annotations.Nullable;
```
Comment (isper3at) [inline extras/rya.console/src/main/java/org/apache/rya/shell/SharedShellState.java]: hurmmm?  this intentional?

File: extras/rya.pcj.fluo/pcj.fluo.app/src/main/java/org/apache/rya/indexing/pcj/fluo/app/query/FilterMetadata.java [hunk 2]
```diff
@@ -21,9 +21,9 @@
 import static com.google.common.base.Preconditions.checkArgument;
 import static com.google.common.base.Preconditions.checkNotNull;
 
-import javax.annotation.Nullable;
-import javax.annotation.ParametersAreNonnullByDefault;
-import javax.annotation.concurrent.Immutable;
+import edu.umd.cs.findbugs.annotations.Nullable;
```
Comment (isper3at) [inline extras/rya.pcj.fluo/pcj.fluo.app/src/main/java/org/apache/rya/indexing/pcj/fluo/app/query/FilterMetadata.java]: Here too

File: extras/rya.pcj.fluo/pcj.fluo.app/src/main/java/org/apache/rya/indexing/pcj/fluo/app/query/FluoQuery.java [hunk 3]
```diff
@@ -25,9 +25,9 @@
 import java.util.Map;
 import java.util.Map.Entry;
 
-import javax.annotation.Nullable;
-import javax.annotation.ParametersAreNonnullByDefault;
-import javax.annotation.concurrent.Immutable;
+import edu.umd.cs.findbugs.annotations.Nullable;
```
Comment (isper3at) [inline extras/rya.pcj.fluo/pcj.fluo.app/src/main/java/org/apache/rya/indexing/pcj/fluo/app/query/FluoQuery.java]: and here?

File: extras/rya.pcj.fluo/pcj.fluo.app/src/main/java/org/apache/rya/indexing/pcj/fluo/app/query/JoinMetadata.java [hunk 4]
```diff
@@ -20,9 +20,9 @@
 
 import static com.google.common.base.Preconditions.checkNotNull;
 
-import javax.annotation.Nullable;
-import javax.annotation.ParametersAreNonnullByDefault;
-import javax.annotation.concurrent.Immutable;
+import edu.umd.cs.findbugs.annotations.Nullable;
```
Comment (isper3at) [inline extras/rya.pcj.fluo/pcj.fluo.app/src/main/java/org/apache/rya/indexing/pcj/fluo/app/query/JoinMetadata.java]: there are a few of these aren't there

File: extras/rya.pcj.fluo/pcj.fluo.app/src/main/java/org/apache/rya/indexing/pcj/fluo/app/query/QueryMetadata.java [hunk 5]
```diff
@@ -20,9 +20,9 @@
 
 import static com.google.common.base.Preconditions.checkNotNull;
 
-import javax.annotation.Nullable;
-import javax.annotation.ParametersAreNonnullByDefault;
-import javax.annotation.concurrent.Immutable;
+import edu.umd.cs.findbugs.annotations.Nullable;
```
Comment (isper3at) [inline extras/rya.pcj.fluo/pcj.fluo.app/src/main/java/org/apache/rya/indexing/pcj/fluo/app/query/QueryMetadata.java]: ping

File: extras/rya.pcj.fluo/pcj.fluo.app/src/main/java/org/apache/rya/indexing/pcj/fluo/app/query/StatementPatternMetadata.java [hunk 6]
```diff
@@ -20,9 +20,9 @@
 
 import static com.google.common.base.Preconditions.checkNotNull;
 
-import javax.annotation.Nullable;
-import javax.annotation.ParametersAreNonnullByDefault;
-import javax.annotation.concurrent.Immutable;
+import edu.umd.cs.findbugs.annotations.Nullable;
```
Comment (isper3at) [inline extras/rya.pcj.fluo/pcj.fluo.app/src/main/java/org/apache/rya/indexing/pcj/fluo/app/query/StatementPatternMetadata.java]: pong

File: extras/rya.pcj.fluo/pcj.fluo.client/src/main/java/org/apache/rya/indexing/pcj/fluo/client/PcjAdminClientProperties.java [hunk 7]
```diff
@@ -22,14 +22,14 @@
 
 import java.util.Properties;
 
-import javax.annotation.Nullable;
-import javax.annotation.ParametersAreNonnullByDefault;
+import edu.umd.cs.findbugs.annotations.Nullable;
```
Comment (isper3at) [inline extras/rya.pcj.fluo/pcj.fluo.client/src/main/java/org/apache/rya/indexing/pcj/fluo/client/PcjAdminClientProperties.java]: pang

**Comment by isper3at** [comment]
Just the Nullable annotation.  I imagine it has something to do with findbugs, but it seems odd that we would have to use the 3rd party annotation over the java one.  I kinda wish we could link the // SEE RYA-211 with an actual javadoc link to jira

File: extras/rya.console/src/main/java/org/apache/rya/shell/SharedShellState.java (hunk 1 above)
Comment (amihalik) [inline extras/rya.console/src/main/java/org/apache/rya/shell/SharedShellState.java]: Do you mean the new package name? Yeah, that's intentional but I don't know if it's correct.  I was told to use findbug-annotations, and that's where the nullable annotation lives.

**Comment by joshelser** [comment]
ThreadSafe and Immutable are available in https://github.com/stephenc/jcip-annotations

**Comment by joshelser** [comment]
Not sure about ParametersAreNonNullByDefault, curiously, it's mentioned https://github.com/stephenc/findbugs-annotations/blob/master/src/spec/javadoc/edu/umd/cs/findbugs/annotations/DefaultAnnotationForParameters.html#L87-L91

**Comment by amihalik** [comment]
@joshelser thanks a lot for finding those libraries and reading the docs in detail to find the equivalent to ParametersAreNonNullByDefault.

I've updated the PR with your findings, and if there are no more comments, I'll pull it into master tomorrow.  (Note that the Jenkins failure is the result of some test failures that I've fixing in another PR).

**Pull request closed without merging**